client/lib/date/index.js

```diff
@@ -48,6 +48,16 @@ export const periods = [
 	{ value: 'previous_year', label: __( 'Previous Year', 'wc-admin' ) },
 ];
 
+/**
+ * Adds timestamp to a string date.
+ *
+ * @param {string} date - Date as a string.
+ * @param {string} timeOfDay - Either `start` or `end` of the day.
+ * @return {string} - String date with timestamp attached.
+ */
+export const appendTimestamp = ( date, timeOfDay ) =>
+	timeOfDay === 'start' ? date + 'T00:00:00+00:00' : date + 'T23:59:59+00:00';
```

Comment:
Just out of curiousity here, do we always want to use `T23:59:59+00:00` as a default?  Are there any other dates that we'll need to use here?

My only concern would be that misspelling `start` (e.g., using `startt`) may make it more difficult to find issues.

Reply:
That's a good point. Probably defaulting to `T23:59:59+00:00` was not the best option.

I wonder what would be the best thing to do if `timeOfDay` is not one of the valid options: throwing an error? returning the `date` without appending any timestamp?

For now, I implemented it so it throws an error if `timeOfDay` is not `start` or `end`. Let me know how it looks! :slightly_smiling_face:

Reply:
I think throwing an error was the best idea since we're expecting a specific timestamp from a string.  Looks good!